modular_skyrat/modules/modular_implants/code/nifsofts/scryer.dm

```diff
@@ -0,0 +1,105 @@
+/obj/item/disk/nifsoft_uploader/scryer
+	name = "NIFSoft Scryer Uploader Disk"
+	loaded_nifsoft = /datum/nifsoft/scryer
+
+/datum/nifsoft/scryer
+	name = "Scryer"
+	program_desc = "Allows the user to summon a MODSuit Scryer"
+	active_mode = TRUE
+	active_cost = 1
+	activation_cost = 20
+	purchase_price = 200
+	buying_category = NIFSOFT_CATEGORY_UTILITY
+	ui_icon = "video"
+	/// What is the scryer currently associated with the NIFSoft?
+	var/obj/item/clothing/neck/link_scryer/loaded/nifsoft/linked_scyer
+
+/datum/nifsoft/scryer/New()
+	. = ..()
+	linked_scyer = new (parent_nif.resolve())
+	linked_scyer.parent_nifsoft = WEAKREF(src)
+	linked_scyer.label = linked_mob.name
+
+/datum/nifsoft/scryer/Destroy()
+	if(linked_scyer)
+		qdel(linked_scyer)
+		linked_scyer = null
+
+	return ..()
+
+/datum/nifsoft/scryer/activate()
+	. = ..()
+	if(. == FALSE)
+		return FALSE
+
+	if(!active)
+		if(linked_scyer)
+			linked_mob.transferItemToLoc(linked_scyer, parent_nif.resolve(), TRUE)
+		return TRUE
+
+	if(linked_mob.handcuffed)
+		linked_mob.balloon_alert(linked_mob, "handcuffed")
+		activate()
+		return FALSE
+
+	if(!linked_mob.equip_to_slot_if_possible(linked_scyer, ITEM_SLOT_NECK)) //This sends out a message to the mob if it can't be put on.
+		activate()
+		return FALSE
+
+	return TRUE
+
+/obj/item/clothing/neck/link_scryer
+	/// Do we have custom controls? This is only affects the text shown when examining
+	var/custom_examine_controls = FALSE
+
+/obj/item/clothing/neck/link_scryer/loaded/nifsoft
+	name = "\improper NIFSoft scryer"
```

Comment:
NIFLink Holocaller

desc = "A nanomachine construct working as a modified version of the MODlink scryer, conjured using a NIF; functionally the same, but able to carry out holocalls in a more portable format."